semanticcpg/src/main/scala/io/shiftleft/semanticcpg/accesspath/TrackedBase.scala

```diff
@@ -14,12 +14,22 @@ case class TrackedLiteral(literal: Literal) extends TrackedBase {
     s"TrackedLiteral(${literal.code})"
   }
 }
-case class TrackedMethodOrTypeRef(methodOrTypeRef: StoredNode with HasCode) extends TrackedBase {
+
+sealed trait TrackedMethodOrTypeRef extends TrackedBase {
+  def code: String
+
   override def toString: String = {
-    s"TrackedMethodOrTypeRef(${methodOrTypeRef.code})"
+    s"TrackedMethodOrTypeRef($code)"
   }
 }
 
+case class TrackedMethod(method: MethodRef) extends TrackedMethodOrTypeRef {
+  override def code = method.code
+}
+case class TrackedTypeRef(typeRef: TypeRef) extends TrackedMethodOrTypeRef {
+  override def code = typeRef.code
+}
+
```

Comment:
This would require downstream changes in the data flow tracker. It is not much of a deal but i would like to suggest using alternative constructors instead. So that we would have:
```
case class TrackedMethodOrTypeRef(methodOrTypeRef: StoredNode, code: String) {
  def this(methodRef: MethodRef) ...
 def this(typeRef: TypeRef)...
}

Reply:
mmh, afaics all affected places downstream are actually beneficial... e.g. 
```
.collect { case TrackedMethodOrTypeRef(methodRef: MethodRef) => methodRef }
->
.collect { case TrackedMethod(methodRef) => methodRef }
```

another example
```
        .collectAll[TrackedMethodOrTypeRef]
        .foreach {
          case TrackedMethodOrTypeRef(ref: MethodRef) =>
            ...
          case _ =>

->
        .collectAll[TrackedMethod].foreach { ref => 
           ...
        }
```

Reply:
Hmm yeah, there are cases where the new representation is better and in others the old. So i would rather stick to what it is.

Reply:
resolved via zoom - we'll take the new representation